src/index.js

```diff
@@ -1,52 +1,125 @@
 'use strict';
 
+const fs = require('fs');
+const FreeSurfer = require('freesurfer-parser');
 const pkg = require('../package.json');
 
+/**
+ * Get FreeSurfer region of interest picker.
+ *
+ * @param {string} feature
+ * @returns {Function} FreeSurfer data picker
+ */
+const getFreeSurferDataPicker = feature => freeSurferData => {
+  return new FreeSurfer({ string: freeSurferData })[feature];
+};
+
+/**
+ * Read file.
+ *
+ * @param {string} file
+ * @returns {Promise} Resolves to `file`'s contents
+ */
+const readFileAsync = file => new Promise((resolve, reject) => {
+  fs.readFile(file, (error, data) => {
+    if (error) {
+      reject(error);
+    } else {
+      resolve(data.toString());
+    }
+  });
+});
+
+/**
+ * Get normalized tags.
+ *
+ * @param {Object} file
+ * @param {Object} file.tags Hash map of tag names to values
+ * @returns {Array} Normalized tags
+ */
+const getNormalizedTags = file => Object.keys(file.tags).sort().reduce(
+  (memo, tag) => {
+    const value = file.tags[tag];
+
+    if (typeof value === 'boolean') {
+      return memo.concat(value === true ? 1 : -1);
+    } else if (typeof value === 'number') {
+      return memo.concat(value);
+    }
+
+    return memo;
+  },
+  []
+);
+
 module.exports = {
   name: pkg.name,
   version: pkg.version,
-  local: {
+  local: [{
+    type: 'function',
+    fn(opts) {
+      let features = ((((opts.remoteResult || {}).pluginState || {}).inputs || [])[0] || [])[0];
+      const files = (opts.userData || {}).files;
+
+      // TODO: This is a hack for simulator. Figure out how to load plugin state
+      if (!features && opts.userData.__FEATURES__) {
+        features = opts.userData.__FEATURES__;
+      }
+
+      if (!Array.isArray(features) || !features.length) {
+        return Promise.reject(new Error('Expected RoI features inputs'));
+      } else if (features.some(f => FreeSurfer.validFields.indexOf(f) < 0)) {
+        return Promise.reject(new Error(
+          `Invalid FreeSurfer feature in ${features.toString()}`
+        ));
+      } else if (!Array.isArray(files) || !files.length) {
+        return Promise.reject(new Error('Expected user data to contain files'));
+      }
+
+      const pickFeature = getFreeSurferDataPicker(features[0]);
+
+      return Promise.all(files.map(f => readFileAsync(f.filename)))
+        .then(freeSurferDatas => ({
+          X: files.map(getNormalizedTags), // covariates
+          y: freeSurferDatas.map(pickFeature), // RoIs
+        }));
+    },
+    inputs: [{
+      help: 'Select Freesurfer region(s) of interest',
+      label: 'Freesurfer ROI',
+      type: 'select',
+      values: FreeSurfer.validFields,
+    }, {
+      label: 'Covariates',
+      type: 'covariates',
+    }],
+  }, {
     type: 'cmd',
     cmd: 'python',
     args: ['./ridge_regress.py'],
     verbose: true,
-  },
+  }],
   remote: {
     type: 'function',
     fn(opts) {
-      const data = opts.previousData || {};
```

Comment:
A lot of this can be cleaned up to eliminate unused variables, iterations and `console` calls.